### PR TITLE
Fix Blackout Bug + Typo

### PR DESCRIPTION
**Pull request by codeacula**
This PR is based on PR #84, so that one should probably be merged in first

Ran into this while hunting:

image

So I added some guards to these specific spots. Also fixed a typo in a variable name.

### CODE REVIEW

File: script.lua [hunk 1]
```diff
@@ -1367,15 +1378,18 @@ keneanung.bashing.vitalsChangeRecord = function()
 
 	if keneanung.bashing.attacking == 0 then return end
 
-	local difference = keneanung.bashing.lastHealth - gmcp.Char.Vitals.hp
+  if gmcp.Char.Vitals.hp then
```
Comment (keneanung) [inline script.lua:1381]:
```suggestion
	if gmcp.Char.Vitals.hp then
```

**Review by keneanung — APPROVED** [review]
While I'm not 100% happy with this solution, I agree that I put this off for far too long. This implementation will skew the heuristics for fleeing quite a bit, but I don't have anything better. Let's merge it in and have a look how people are perceiving the change.

I have just one little nitpick for formatting.

File: script.lua (hunk 1 above)
Comment (codeacula) [reply script.lua:1381]: It looks like you want this indented, but why? It's not inside a control structure.

**Comment by codeacula** [comment]
I'm curious how it will mess up with the fleeing heuristics, as you say. All this should be doing is not allow code that would normally error to process.

**Comment by codeacula** [comment]
What do you think about using `keneanung.bashing.lastHealth` instead of the current HP? My solution right now appears to be causing issues when the player blacks out. It stops bashing

**Comment by codeacula** [comment]
I switched the guard so instead of trying to use GMCP it uses that last known health value. That's a little better than what was originally there (because it doesn't error out), and doesn't mess up bashing by bypassing the whole statement and skipping to clear queue

**Comment by keneanung** [comment]
The heuristic screw up happens as attacks are recorded, but health changes are not, effectively lowering the average damage received per attack. But as I said, I have no better idea myself, how to improve the system.

The solution looks good overall and is not really a change in how the values are accessed. But I must admit that the naming `lastHealth` threw me off here. It sounds as if the prompt's health value before the attack. That would not be the right thing to use.

File: script.lua (hunk 1 above)
Comment (keneanung) [reply script.lua:1383]: But it's inside a function. And indentation is done with tabs, not spaces in this project. I'd do this differently nowadays, but for consistency's sake, let's stick with a single tab per level.

Comment (codeacula) [reply script.lua:1383]: Ohhh got ya!

**Comment by codeacula** [comment]
`lastHealth` was something that was already in there. You put it in there around 8 years ago 😁. You're also doing this with it, which I thought was weird but maybe there's a reason?

`keneanung.bashing.lastHealth = gmcp.Char.Vitals.hp * 1`

Anyway, I think I addressed the changes

File: script.lua [hunk 2]
```diff
@@ -1378,15 +1380,18 @@ keneanung.bashing.vitalsChangeRecord = function()
...
+		local difference = keneanung.bashing.lastHealth - gmcp.Char.Vitals.hp
 
-	keneanung.bashing.lastHealth = gmcp.Char.Vitals.hp * 1
+		if difference > 0 then
+			keneanung.bashing.damage = keneanung.bashing.damage + difference
+		elseif difference < 0 then
+			keneanung.bashing.healing = keneanung.bashing.healing + math.abs(difference)
+		end
+
+		keneanung.bashing.lastHealth = gmcp.Char.Vitals.hp * 1.
```
Comment (keneanung) [inline script.lua:1393]: is the period at the end of the line here by design?

Comment (codeacula) [reply script.lua]: Resolved!

**Comment by codeacula** [comment]
Bumping this. Is there anything else you need from me for it?

**Review by keneanung — APPROVED** [review]

**Comment by keneanung** [comment]
Sorry for the long wait. I haven't been in the right headspace and it fell off the radar...

**Pull request merged**